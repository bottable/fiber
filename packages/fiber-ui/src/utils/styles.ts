import {
  compose,
  space,
  color,
  typography,
  height,
  SpaceProps,
  ColorProps,
  TypographyProps
} from 'styled-system'

export const styleComposition = compose(space, height, color, typography)

export type StyleProps = SpaceProps & ColorProps & TypographyProps
