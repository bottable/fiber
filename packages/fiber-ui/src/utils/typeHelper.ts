export function withProperties<A, B>(component: A, properties: B): A & B {
  Object.keys(properties).forEach((key) => {
    // @ts-ignore
    component[key] = properties[key]
  })
  return component as A & B
}
