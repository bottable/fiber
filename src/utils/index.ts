export * from './typeHelper'
export * from './mergeProps'
