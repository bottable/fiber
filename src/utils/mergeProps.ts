import React from 'react'

export type MergeElementProps<
  T extends React.ElementType,
  P extends object = {}
> = Omit<React.ComponentPropsWithRef<T>, keyof P> & P
