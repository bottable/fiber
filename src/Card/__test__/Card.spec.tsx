import { Card } from '..'

import React from 'react'
import { render } from 'test/utils'
import SettingsIcon from '@material-ui/icons/Settings'
import CreateIcon from '@material-ui/icons/Create'
import MoreHorizIcon from '@material-ui/icons/MoreHoriz'

describe('Card', () => {
  it('renders correctly', () => {
    const card = render(
      <>
        <div>
          <Card style={{ width: 300 }}>
            <p>Card content</p>
            <p>Card content</p>
            <p>Card content</p>
          </Card>
        </div>
        <div>
          <Card
            title='Default size card'
            extra={<a href='#'>More</a>}
            style={{ width: 300 }}
          >
            <p>Card content</p>
            <p>Card content</p>
            <p>Card content</p>
          </Card>
        </div>
        <div>
          <Card title='Medium card' style={{ width: 300 }}>
            <p style={{ margin: 0 }}>Card content</p>
            <p style={{ margin: 0 }}>Card content</p>
            <p style={{ margin: 0 }}>Card content</p>
          </Card>
          <br />
          <Card title='Small card' size='sm' style={{ width: 300 }}>
            <p style={{ margin: 0 }}>Card content</p>
            <p style={{ margin: 0 }}>Card content</p>
            <p style={{ margin: 0 }}>Card content</p>
          </Card>
        </div>
        <div>
          <div
            style={{
              width: 300,
              background: 'rgb(236, 236, 236)',
              padding: 30
            }}
          >
            <Card title='No borders' style={{ width: 300 }} bordered={false}>
              <p style={{ margin: 0 }}>Card content</p>
              <p style={{ margin: 0 }}>Card content</p>
              <p style={{ margin: 0 }}>Card content</p>
            </Card>
          </div>
        </div>
        <div>
          <Card title='Hoverable' style={{ width: 300 }} hoverable>
            <p style={{ margin: 0 }}>Card content</p>
            <p style={{ margin: 0 }}>Card content</p>
            <p style={{ margin: 0 }}>Card content</p>
          </Card>
        </div>
        <div>
          <Card
            title='Actions'
            style={{ width: 300 }}
            actions={[
              <SettingsIcon key={1} />,
              <CreateIcon key={2} />,
              <MoreHorizIcon key={3} />
            ]}
          >
            <p style={{ margin: 0 }}>Card content</p>
            <p style={{ margin: 0 }}>Card content</p>
            <p style={{ margin: 0 }}>Card content</p>
          </Card>
        </div>
      </>
    )

    expect(card).toMatchSnapshot()
  })
})
