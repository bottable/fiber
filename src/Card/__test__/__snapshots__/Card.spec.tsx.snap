// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders correctly 1`] = `
Array [
  <div>
    <div
      className="sc-AxjAm cLazpZ"
      style={
        Object {
          "width": 300,
        }
      }
    >
      <div
        className="sc-AxhUy gHKwLs"
      >
        <p>
          Card content
        </p>
        <p>
          Card content
        </p>
        <p>
          Card content
        </p>
      </div>
    </div>
  </div>,
  <div>
    <div
      className="sc-AxjAm cLazpZ"
      style={
        Object {
          "width": 300,
        }
      }
    >
      <div
        className="sc-AxirZ hpwXmn"
      >
        <div
          className="sc-AxiKw PyuZw"
        >
          Default size card
        </div>
        <div
          className="sc-AxhCb ekOLyM"
        >
          <a
            href="#"
          >
            More
          </a>
        </div>
      </div>
      <div
        className="sc-AxhUy gHKwLs"
      >
        <p>
          Card content
        </p>
        <p>
          Card content
        </p>
        <p>
          Card content
        </p>
      </div>
    </div>
  </div>,
  <div>
    <div
      className="sc-AxjAm cLazpZ"
      style={
        Object {
          "width": 300,
        }
      }
    >
      <div
        className="sc-AxirZ hpwXmn"
      >
        <div
          className="sc-AxiKw PyuZw"
        >
          Medium card
        </div>
      </div>
      <div
        className="sc-AxhUy gHKwLs"
      >
        <p
          style={
            Object {
              "margin": 0,
            }
          }
        >
          Card content
        </p>
        <p
          style={
            Object {
              "margin": 0,
            }
          }
        >
          Card content
        </p>
        <p
          style={
            Object {
              "margin": 0,
            }
          }
        >
          Card content
        </p>
      </div>
    </div>
    <br />
    <div
      className="sc-AxjAm cLazpZ"
      size="sm"
      style={
        Object {
          "width": 300,
        }
      }
    >
      <div
        className="sc-AxirZ jNtecM"
        size="sm"
      >
        <div
          className="sc-AxiKw giBzay"
          size="sm"
        >
          Small card
        </div>
      </div>
      <div
        className="sc-AxhUy hxqVyW"
        size="sm"
      >
        <p
          style={
            Object {
              "margin": 0,
            }
          }
        >
          Card content
        </p>
        <p
          style={
            Object {
              "margin": 0,
            }
          }
        >
          Card content
        </p>
        <p
          style={
            Object {
              "margin": 0,
            }
          }
        >
          Card content
        </p>
      </div>
    </div>
  </div>,
  <div>
    <div
      style={
        Object {
          "background": "rgb(236, 236, 236)",
          "padding": 30,
          "width": 300,
        }
      }
    >
      <div
        className="sc-AxjAm dbubJq"
        style={
          Object {
            "width": 300,
          }
        }
      >
        <div
          className="sc-AxirZ hpwXmn"
        >
          <div
            className="sc-AxiKw PyuZw"
          >
            No borders
          </div>
        </div>
        <div
          className="sc-AxhUy gHKwLs"
        >
          <p
            style={
              Object {
                "margin": 0,
              }
            }
          >
            Card content
          </p>
          <p
            style={
              Object {
                "margin": 0,
              }
            }
          >
            Card content
          </p>
          <p
            style={
              Object {
                "margin": 0,
              }
            }
          >
            Card content
          </p>
        </div>
      </div>
    </div>
  </div>,
  <div>
    <div
      className="sc-AxjAm bSDCLu"
      style={
        Object {
          "width": 300,
        }
      }
    >
      <div
        className="sc-AxirZ hpwXmn"
      >
        <div
          className="sc-AxiKw PyuZw"
        >
          Hoverable
        </div>
      </div>
      <div
        className="sc-AxhUy gHKwLs"
      >
        <p
          style={
            Object {
              "margin": 0,
            }
          }
        >
          Card content
        </p>
        <p
          style={
            Object {
              "margin": 0,
            }
          }
        >
          Card content
        </p>
        <p
          style={
            Object {
              "margin": 0,
            }
          }
        >
          Card content
        </p>
      </div>
    </div>
  </div>,
  <div>
    <div
      className="sc-AxjAm cLazpZ"
      style={
        Object {
          "width": 300,
        }
      }
    >
      <div
        className="sc-AxirZ hpwXmn"
      >
        <div
          className="sc-AxiKw PyuZw"
        >
          Actions
        </div>
      </div>
      <div
        className="sc-AxhUy gHKwLs"
      >
        <p
          style={
            Object {
              "margin": 0,
            }
          }
        >
          Card content
        </p>
        <p
          style={
            Object {
              "margin": 0,
            }
          }
        >
          Card content
        </p>
        <p
          style={
            Object {
              "margin": 0,
            }
          }
        >
          Card content
        </p>
      </div>
      <div
        className="sc-AxgMl fZFoSa"
      >
        <div
          className="sc-AxheI igKBMW"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.95 10.78c.03-.25.05-.51.05-.78s-.02-.53-.06-.78l1.69-1.32c.15-.12.19-.34.1-.51l-1.6-2.77c-.1-.18-.31-.24-.49-.18l-1.99.8c-.42-.32-.86-.58-1.35-.78L12 2.34c-.03-.2-.2-.34-.4-.34H8.4c-.2 0-.36.14-.39.34l-.3 2.12c-.49.2-.94.47-1.35.78l-1.99-.8c-.18-.07-.39 0-.49.18l-1.6 2.77c-.1.18-.06.39.1.51l1.69 1.32c-.04.25-.07.52-.07.78s.02.53.06.78L2.37 12.1c-.15.12-.19.34-.1.51l1.6 2.77c.1.18.31.24.49.18l1.99-.8c.42.32.86.58 1.35.78l.3 2.12c.04.2.2.34.4.34h3.2c.2 0 .37-.14.39-.34l.3-2.12c.49-.2.94-.47 1.35-.78l1.99.8c.18.07.39 0 .49-.18l1.6-2.77c.1-.18.06-.39-.1-.51l-1.67-1.32zM10 13c-1.65 0-3-1.35-3-3s1.35-3 3-3 3 1.35 3 3-1.35 3-3 3z"
              transform="scale(1.2, 1.2)"
            />
          </svg>
        </div>
        <div
          className="sc-AxheI igKBMW"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
            />
          </svg>
        </div>
        <div
          className="sc-AxheI igKBMW"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M6 10c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm12 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm-6 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>,
]
`;
